### ISSUE TITLE
channels won't go active due to some htlc things

### ISSUE THREAD

**Issue by robtex** [lightningnetwork/lnd#1642]
I have a few channels that won't go active.
when i try to connect to them i get something like this:
seems to happen with both CL and lnd peers
`lnd version 0.4.2-beta commit=098cd940e312c111dda3efe5fa2c7d375743f770`

```
2018-07-27 20:23:24.236 [WRN] SRVR: Already have 1 persistent connection requests for 038118e0490f8d3df07b6bdc5e7034dee90d769012764d31bfeb167269866824f0@35.185.13.214:9735, connecting anyway.                                                           
2018-07-27 20:23:24.277 [INF] SRVR: Established connection to: 35.185.13.214:9735
2018-07-27 20:23:24.277 [INF] SRVR: finalizing connection to 038118e0490f8d3df07b6bdc5e7034dee90d769012764d31bfeb167269866824f0, inbound=false                                                                                                           
2018-07-27 20:23:24.304 [INF] PEER: NodeKey(038118e0490f8d3df07b6bdc5e7034dee90d769012764d31bfeb167269866824f0) loading ChannelPoint(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1)
2018-07-27 20:23:24.314 [DBG] CNCT: New ChainEventSubscription(id=43) for ChannelPoint(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1)
2018-07-27 20:23:24.315 [INF] HSWC: ChannelLink(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1) is starting                                                                                                                           
2018-07-27 20:23:24.315 [INF] HSWC: Trimming open circuits for chan_id=528615:1949:1, start_htlc_id=84                       
2018-07-27 20:23:24.315 [INF] HSWC: Adding live link chan_id=dcd4e12efa5ea08c01d672c25818486c3eca2341fa555a3c0de522c9cfc7d5ec, short_chan_id=528615:1949:1                                                                                                
2018-07-27 20:23:24.315 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1)
2018-07-27 20:23:24.316 [TRC] CNCT: ChannelArbitrator(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1) got new signal update!
2018-07-27 20:23:24.315 [INF] HSWC: HTLC manager for ChannelPoint(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1) started, bandwidth=622375233 mSAT                                                                                   
2018-07-27 20:23:24.316 [INF] HSWC: Attempting to re-resynchronize ChannelPoint(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1)
2018-07-27 20:23:24.365 [INF] HSWC: Received re-establishment message from remote side for channel(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1)                                                                                   
2018-07-27 20:23:24.871 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed                                                                                             
2018-07-27 20:23:24.871 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed
2018-07-27 20:23:24.871 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed
2018-07-27 20:23:24.871 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed
2018-07-27 20:23:24.871 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed
2018-07-27 20:23:24.872 [ERR] HSWC: ChannelLink(528615:1949:1) unhandled error while forwarding htlc packet over htlcswitch: circuit has already been closed
2018-07-27 20:23:25.255 [ERR] HSWC: ChannelLink(528615:1949:1) Failing link: unable to settle incoming HTLC: No HTLC with ID 171 in channel 528615:1949:1
2018-07-27 20:23:25.255 [ERR] HSWC: ChannelLink(528615:1949:1) link failed, exiting htlcManager                              
2018-07-27 20:23:25.255 [INF] HSWC: ChannelLink(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1) has exited
2018-07-27 20:23:25.255 [INF] HSWC: Removing channel link with ChannelID(dcd4e12efa5ea08c01d672c25818486c3eca2341fa555a3c0de522c9cfc7d5ec)
2018-07-27 20:23:25.255 [INF] PEER: unable to read message from 35.185.13.214:9735: read tcp 167.99.50.31:50248->35.185.13.214:9735: use of closed network connection
2018-07-27 20:23:25.255 [INF] HSWC: ChannelLink(edd5c7cfc922e50d3c5a55fa4123ca3e6c481858c272d6018ca05efa2ee1d4dc:1) is stopping
2018-07-27 20:23:25.256 [INF] DISC: Removing gossipSyncer for peer=038118e0490f8d3df07b6bdc5e7034dee90d769012764d31bfeb167269866824f0
2018-07-27 20:23:25.256 [ERR] SRVR: unable to get channel links: unable to locate channel link by destination hop id 038118e0490f8d3df07b6bdc5e7034dee90d769012764d31bfeb167269866824f0
```

**Comment by cfromknecht** [comment]
thanks @robtex! i worked on some fixes for this issue last night

in the meantime you can comment out this line, as this occurrence shouldn't be treated as a link failure https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L1149
you can replace it with
```
fmt.Errorf("unable to settle incoming HTLC: %v", err)
```
if you still wanna see the error

**Comment by robtex** [comment]
and keep the return?
i'll check what it says, i got tired of the errors so i might have close most of those channels, but we will see.
discovered it because ppl tried to route thru that channel, thinking it was active.

**Comment by cfromknecht** [comment]
yeah this happens when the packet has already been processed, and internally we're reforwarding a duplicate. keep the return, just want to drop the packet.

**Comment by Roasbeef** [comment]
This issue is: 
```
2018-07-27 20:23:25.255 [ERR] HSWC: ChannelLink(528615:1949:1) Failing link: unable to settle incoming HTLC: No HTLC with ID 171 in channel 528615:1949:1
2018-07-27 20:23:25.255 [ERR] HSWC: ChannelLink(528615:1949:1) link failed, exiting htlcManager                              
```

The remote side is trying to settle an HTLC that we don't actually have. It's a violation on their side even.

**Comment by cfromknecht** [comment]
Nope that's an internal retransmission, other side is likely fine. Currently we fail the channel for duplicate retransmission of settles, but not fails. My WIP branch with a fix for this is here: https://github.com/cfromknecht/lnd/tree/switch-fwdpkg-cleanup

**Comment by cfromknecht** [comment]
Resolved by #1698, #1747, #1748, #1166, #1551, #1658 and a bunch more
